### ISSUE TITLE
NVDA does not honour "Automatic focus mode for focus changes" when using the Elements List dialog "Move to" button

### ISSUE THREAD

**Issue by JulienCochuyt** [nvaccess/nvda#10455]
### Steps to reproduce:

In Preferences > Browse mode, ensure "Automatic focus mode for focus changes" is checked.
Open any document - such as a web page - containing a focusable form field.
Ensure you are in browse mode.
Open the Elements List dialog using `NVDA+F7`.
Select "Form fields" type.
Select the focusable form field in the list.
Activate the "Move to" button.

### Actual behavior:

Focus moved to the form field, but NVDA is still in browse mode.

### Expected behavior:

NVDA should be in focus mode.

#### NVDA installed/portable/running from source:

Installed

#### NVDA version:

NVDA 2019.2.1

#### Windows version:

Windows 10 version 1903 (18362.418)

#### Name and version of other software in use when reproducing the issue:

Google Chrome 78

#### Other information about your system:

N/A

#### Does the issue still occur after restarting your PC?

Yes.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

No.

**Comment by Adriani90** [comment]
This also happens in Firefox 127 with NVDA 2024.2, and it also happens when pressing enter on the form field to which the user wants to navigate. It is not only related to the "move to button".

e.g.
1. Make sure automatic focus mode both for system focus and for system caret are enabled in NVDA browse mode settings
2. On this page, press NVDA+f7
3. Pres shift+tab and use arrow keys to switch to form fields
4. In the tree view, arrow down to the comment edit field and press enter on it

Actual:
The virtual cursor jumps to the comment edit field, but focus mode is not enabled automatically. You have to press arrow up and down to trigger the automatic focus mode.

Expected:
Automatic focus mode should trigger directly when the virtual cursor reaches the element where this is supported, such as an edit field, even wehn the user navigates to it out of the elements list.

cc: @jcsteh at least for Firefox. But maybe this has to do with the automatic moving the system focus / caret to focusable elements out of browse mode. I guess the elements list use case is not considered yet.
